lib/collection/src/shards/replica_set/partial_snapshot_meta.rs

```diff
@@ -88,44 +92,10 @@ impl PartialSnapshotMeta {
             .unwrap_or(Duration::ZERO)
             .as_secs();
 
-        self.recovery_timestamp
-            .store(timestamp, atomic::Ordering::Relaxed);
+        self.recovery_timestamp.store(timestamp, Ordering::Relaxed);
     }
 }
 
 fn recovery_in_progress() -> CollectionError {
     CollectionError::shard_unavailable("partial snapshot recovery is in progress")
 }
-
-#[derive(Debug, Default)]
-pub struct RequestTracker {
```

Comment:
I noticed a bit too late that there is already some sort of `ScopeTracker`. I decided to keep my implementation because it is located at a more generic location, has tests and is implemented a bit more generic.